bitcoin/src/blockdata/script/mod.rs

```diff
@@ -86,9 +86,9 @@ pub use self::{
 
 hashes::hash_newtype! {
     /// A hash of Bitcoin Script bytecode.
-    pub struct ScriptHash(hash160::Hash);
+    pub struct ScriptHash(hash160);
```

Comment:
Oh, interesting. It limits the macro to our library but that's most likely OK.

Reply:
Oh, good catch. No, we definitely do not want to limit the macro to this library. We definitely need it in rust-miniscript, as well as rust-elements.

Reply:
This was awesome, IIUC the initial comment is not correct but in proving so I uncovered a couple of bugs in how the visibility on the inner field was handled. Please see `hashes/examples/*` for usage of both macros.

Please note that the two macros differ in that the `hash_newtype` allows additional attributes on the inner field but the `sha256t_hash_newtype` does not - is that ok?

Reply:
I mean you can't have an external crate defining `FooHash` and use it with this macro.

> but the `sha256t_hash_newtype` does not - is that ok?

I think it's not a big deal since it's quite niche feature anyway. It can be added later. Of course it'd be nice to have it eventually.

Reply:
Oh yes, of course. Thanks for clarifying.